scylla/src/transport/session_test.rs

```diff
@@ -168,7 +187,26 @@ async fn test_prepared_statement() {
         let a = r.columns[0].as_ref().unwrap().as_int().unwrap();
         let b = r.columns[1].as_ref().unwrap().as_int().unwrap();
         let c = r.columns[2].as_ref().unwrap().as_text().unwrap();
-        assert_eq!((a, b, c), (17, 16, &String::from("I'm prepared!!!")))
+        assert_eq!((a, b, c), (17, 16, &String::from("I'm prepared!!!")));
+
+        let mut results_from_manual_paging: Vec<Row> = vec![];
+        let query = Query::new("SELECT a, b, c FROM ks.t2".to_owned()).with_page_size(1);
```

Comment:
Does Scylla honor such small page sizes? Quote from the protocol spec:

```
  - Clients should not rely on the actual size of the result set returned to
    decide if there are more results to fetch or not. Instead, they should always
    check the Has_more_pages flag (unless they did not enable paging for the query
    obviously). Clients should also not assert that no result will have more than
    <result_page_size> results. While the current implementation always respects
    the exact value of <result_page_size>, we reserve the right to return
    slightly smaller or bigger pages in the future for performance reasons.
```

Reply:
It does - Scylla will happily return **less** rows due to performance reasons (or even zero), but it will never return more than requested - the rough idea is that if clients experience performance problems due to too small pages, they should bump the page size themselves. Having a page size of 1 makes very little sense in production environments, but it's useful for testing.